internal/gps/version.go

```diff
@@ -274,6 +282,14 @@ func (v branchVersion) Pair(r Revision) PairedVersion {
 	}
 }
 
+func (v branchVersion) equals(c Constraint) bool {
+	v2, ok := c.(branchVersion)
+	if !ok {
+		return false
+	}
+	return v == v2
```

Comment:
this is one of those cases where we the definition of "equality" is really contextualized by how we use it. the default branch property on `branchVersion` is only set correctly by `sourceMgr`, not by anything that parses manifests. if we rely on this equals operator in the wrong place, then we have problems.

but, perhaps you intend that to be the difference between `equals()` and `Matches()`?